### DISCUSSION TITLE
FATAL ERROR in native method: Bad global or local ref passed to JNI

### DISCUSSION THREAD

**Discussion by hcb-qf** [OpenDDS/OpenDDS#4569] [Q&A]
When I run a java test case TestSubscriber in idea, I get the following error. 

*
WARNING in native method: JNI call made without checking exceptions when required to from CallObjectMethodV
	at DDS.TOPIC_QOS_DEFAULT.get(Native Method)
	at com.dds.demo.TestSubscriber.main(TestSubscriber.java:48)
FATAL ERROR in native method: Bad global or local ref passed to JNI
	at DDS.TOPIC_QOS_DEFAULT.get(Native Method)
	at com.dds.demo.TestSubscriber.main(TestSubscriber.java:48)
*
![param](https://github.com/OpenDDS/OpenDDS/assets/166470060/3b232bd3-792a-4745-9814-2bc43fc38351)

**Comment by jrw972** [comment]
Please compare the code from your test application to the code in the developer's guide, examples, and tests so we can understand what is unique to your situation.

**Reply by hcb-qf** [reply]
Thank you for your answer, the TestSubscriber,TestPublisher and DataReaderListenerImpl are copied from the developer's examples

**Reply by jrw972** [reply]
From the error message, it made it into the C++ implementation but then had issues with a reference it acquired from JNI.  So, my best guess is that it's a class not found error.  Are OpenDDS_DCPS.jar, i2jrt.jar, i2jrt_corba.jar, and tao_java.jar all on the class path?
